harness/assert.js

```diff
@@ -84,7 +84,7 @@ assert.throws = function (expectedErrorConstructor, func, message) {
       message += 'Thrown value was not an object!';
       $ERROR(message);
     } else if (thrown.constructor !== expectedErrorConstructor) {
-      message += 'Expected a ' + expectedErrorConstructor.name + ' but got a ' + thrown.constructor.name;
+      message += 'Expected a ' + expectedErrorConstructor.name + ' but got a ' + thrown.constructor.name + ' (' + thrown.message + ')';
```

Comment:
This whole consequent body only concerns the output when there is a constructor mismatch. If the error constructor is not what you wanted, the message is irrelevant.

Reply:
In fact, this has been discussed in the past and the current message is intentionally ignoring the error message.

assert.throws prioritizes a given custom message to the assertion. This is preferred usage and a good practice we should enforce.

Ideally, each test should distinguish assertions, even if it's minimal, e.g.:

```js
assert.throws(Test262Error, fn, "fn throws Test262Error the first time it's called");
assert.throws(TypeError, fn, "fn throws TypeError the second time it's called");
```

If Test262 relies on error messages, it would be expecting implementations to provide something more than the specification does.